src/tests/ftest/nvme/nvme_health.py

```diff
@@ -0,0 +1,116 @@
+        :avocado: tags=nvme_health
+        """
+        # pylint: disable=attribute-defined-outside-init
+        # pylint: disable=too-many-branches
+        no_of_pools = self.params.get("number_of_pools", '/run/pool/*')
+        #Stop the servers to run SPDK too to get the server capacity
+        self.stop_servers()
+        storage = self.get_server_capacity()
+        self.start_servers()
+
+        single_pool_nvme_size = int((storage * 0.80)/no_of_pools)
+
+        pools = []
+        #Create the Large number of pools
+        for _pool in range(no_of_pools):
+            pool = TestPool(self.context, dmg_command=self.get_dmg_command())
+            pool.get_params(self)
+            #SCM size is 10% of NVMe
+            pool.scm_size.update('{}'.format(int(single_pool_nvme_size * 0.10)))
+            pool.nvme_size.update('{}'.format(single_pool_nvme_size))
+            pool.create()
+            pools.append(pool)
+
+        #initialize the dmg command
+        self.dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        self.dmg.get_params(self)
+        self.dmg.insecure.update(
+            self.server_managers[0].get_config_value("allow_insecure"),
+            "dmg.insecure")
+
+        #List all pools
+        self.dmg.set_sub_command("storage")
+        self.dmg.sub_command_class.set_sub_command("query")
+        self.dmg.sub_command_class.sub_command_class.\
+        set_sub_command("list-pools")
+        for host in self.hostlist_servers:
+            self.dmg.hostlist = host
+            try:
+                result = self.dmg.run()
+            except CommandFailure as error:
+                self.fail("dmg command failed: {}".format(error))
+            #Verify all pools UUID listed as part of query
+            for pool in pools:
+                if pool.uuid.lower() not in result.stdout:
+                    self.fail('Pool uuid {} not found in smd query'
+                              .format(pool.uuid.lower()))
+
+        # Get the device ID from all the servers.
+        device_ids = get_device_ids(self.dmg, self.hostlist_servers)
+
+        # Get the device health
+        for host in device_ids:
+            self.dmg.hostlist = host
+            for _dev in device_ids[host]:
+                try:
+                    result = self.dmg.storage_query_device_health(_dev)
+                except CommandFailure as error:
+                    self.fail("dmg get device states failed {}".format(error))
+                if 'State:NORMAL' not in result.stdout:
+                    self.fail("device {} on host {} is not NORMAL"
+                              .format(_dev, host))
+
+        # Get the nvme-health
+        try:
+            self.dmg.storage_query_nvme_health()
+        except CommandFailure as error:
+            self.fail("dmg nvme-health failed {}".format(error))
+
+        # Destroy the pools
+        print('Pool Destroy')
+        for pool in pools:
+            pool.destroy()
```

Comment:
These lines can be avoided by using `self.pool` instead of `pools`

Reply:
Same as above command